### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by tinakurian**
### What does this PR do?
Updates the README to specify the required set of pre-installed tools in order to `make install` and try out one of the samples.

### What issues does this PR fix or reference?
N/A

### Is it tested? How?
N/A

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -27,6 +27,14 @@ metadata:
 ```
 
 ## Running the controller in a cluster
+### Prerequisites
+- go
+- git
+- kustomize.io
+- sed
+- jq
+- yq (python-yq from https://github.com/kislyuk/yq#installation, other distributions may not work)
```
Comment (sleshchenko) [inline README.md:36]: Thank a lot for this line!!!
When I need to tell people which yq is expected, I each time forget and figure out kislyuk vs mikefarah :-D

File: README.md [hunk 2]
```diff
@@ -27,6 +27,14 @@ metadata:
 ```
 
 ## Running the controller in a cluster
+### Prerequisites
+- go
```
Comment (sleshchenko) [inline README.md:31]: go or docker are not really prerequisites for running the controller in a cluster.
Maybe it makes sense to move Prerequisites on one level up.

**Review by sleshchenko — COMMENTED** [review]
The added info is helpful but the current position is not the best.
I don't have a really good proposal, maybe making it `## Prerequisites` above `## Running the controller in a cluster` would be a bit better.

File: README.md (hunk 2 above)
Comment (tinakurian) [reply README.md:29]: Good point, that makes much more sense! I've updated it here: f2a3664d34a7aa06da995545a87232ca4f344186

**Review by sleshchenko — APPROVED** [review]
please rebase against main branch to get PR check fixed

**Review by JPinkney — APPROVED** [review]

**Review by amisevsk — APPROVED** [review]

**Comment by amisevsk** [comment]
/test v5-devworkspaces-operator-e2e 

y'know, in case changing the readme broke e2e tests :)

**Comment by amisevsk** [comment]
/ok-to-test

but I suspect bigger issues :) I had to add https://github.com/devfile/devworkspace-operator/pull/296/commits/06719896c81a463a0c95b52ee372b2ff68633e49 to my PR to get it to stably run on `crc`

**Comment by sleshchenko** [comment]
I use my administrator power to merge PR without e2e tests since it's only changes in README.md

**Pull request merged**